### DISCUSSION TITLE
Isolate the svg-edit interface

### DISCUSSION THREAD

**Discussion by jggsus88** [SVG-Edit/svgedit#807] [New feature request]
My suggestion could be resumed in two parts:
1. Give the possibility to third party developers to integrate the editor without the interface, having only the canvas and using the editor api from their interface.
2. Generate an angular component to be installed inside an angular project.

## Current behavior

You need all the svg-edit interface in order to integrate it in other projects.

## Desired behavior

Give the possibility to third parties to use the svg edit interface or not.

**Comment by jfhenon** [comment]
I think the first step is to move svgcanvas in its own repository and publish it as a npm package.
Svg-edit (or other third party work) would need to import only this package. This is likely not too difficult to do this but the time consuming part would be to separate all unit tests for svgcanvas (and likely to write additional ones).

**Comment by AndrePessoa** [comment]
@jfhenon, do you still consider to do it? Do you need any help on it? I really believe that this could be a huge improvement on this great project.

**Comment by jfhenon** [comment]
Well, not really a priority for now but a good intermediate step would be to publish svgcanvas from the svgedit repository. Obviously, if you would like to help on this, I would make sure I can review your PR promptly. Thanks.

**Comment by AndrePessoa** [comment]
@jfhenon, sure! But let me read a bit more to understand how complex it can be and feel if I'm ready to do it, ok?

**Comment by jfhenon** [comment]
Hello @AndrePessoa Did you progress about this ?
